board/app/models/thread.php

```diff
@@ -56,67 +83,97 @@ public static function getAll($offset, $limit)
         $threads = array();
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM thread LIMIT {$offset}, {$limit}");
-
-        foreach($rows as $row) 
-        {
+            
+        foreach($rows as $row) {
             $threads[] = new self($row);
         }
-
-    return $threads;
+        return $threads;
     }
 
-    public static function countAll()
+    public static function countAll() 
     {
         $db = DB::conn();
         return (int) $db->value('SELECT COUNT(*) FROM thread');
     }
 
-    public function countComments()
+    public static function countAllThreadByUserId($user_id) 
     {
         $db = DB::conn();
-        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ? ", array($this->id));
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($user_id));
     }
 
+    public static function countAllThreadByCategory($category) 
+    {
+        $db = DB::conn();
+        return (int) $db->value('SELECT COUNT(*) FROM thread WHERE user_id = ?', array($category));
+    }
 
-    public static function get($id)
+    public function countComments() 
+    {
+        $db = DB::conn();
+        return (int) $db->value("SELECT COUNT(*) FROM comment WHERE thread_id = ? ", array($this->id));
+    }
+    
+    public static function get($id) 
     {
         $db = DB::conn();
         $row = $db->row('SELECT * FROM thread WHERE id = ?', array($id));
-        
-            if (!$row) 
-            {
-                throw new RecordNotFoundException('No Record Found');
-            }
-
+            
+        if (!$row) {
+            throw new RecordNotFoundException('No Record Found');
+        }
         return new self($row);
     }
-
-    public function getComments($offset, $limit)
+    
+    public function getComments($offset, $limit) 
     {
         $comments = array();
         $db = DB::conn();
         $rows = $db->rows("SELECT * FROM comment WHERE thread_id = ? ORDER BY created ASC LIMIT {$offset}, {$limit}", array($this->id));
```

Comment:
This should be in the Comment model po :bow: You can just call the Comment's getAll function po instead~